frontend/src/components/ViewTrips/trips-container.js

```diff
@@ -72,8 +73,8 @@ class TripsContainer extends React.Component {
           );
 
           this.setState({ trips: tripsContainer });
-        }, (error) => {
-          let errorElement = getErrorElement(error);
+        }, async (error) => {
```

Comment:
Why do you need the async tag here? the whole function is asynchronous by nature.

Reply:
I thought so too and I had a bug because of it. I would never have seen this if I didn't change the query which required me to create a database index (which caused the error callback to execute).

So because of the structure of `onSnapshot` you have to pass a callback for the error function. Even though componentDidMount is async, it seems that for some reason that if I do not explicitly return a promise from `getErrorElement`, the Promise world gets mad. So in order to use the `await` keyword for `getErrorElement`, I have to make the function async.